### DISCUSSION TITLE
Client proxy settings

### DISCUSSION THREAD

**Discussion by antonrykov** [arthurkushman/php-wss#57] [Q&A] [answered]
Hi!
In README.md 
```
// if proxy settings is of need
$config->setProxy('127.0.0.1', '80');
$config->setProxyAuth('proxyUser', 'proxyPass');
```
But i cant find such functions when i try to use project.
I want to connect to the websocket server (provide as API by another project) via proxy. Can i do this with help of this project?

**Comment by arthurkushman** [comment, marked as answer]
Hi, sure u can do this just like in code snippet from readme
```php
<?php
use WSSC\WebSocketClient;
use WSSC\Components\ClientConfig;

$config = new ClientConfig();
$config->setFragmentSize(8096);
$config->setTimeout(15);
$config->setHeaders([
    'X-Custom-Header' => 'Foo Bar Baz',
]);

// if proxy settings is of need
$config->setProxy('127.0.0.1', '80');
$config->setProxyAuth('proxyUser', 'proxyPass');

$client = new WebSocketClient('ws://localhost:8000/notifications/messanger/yourtoken123', $config);
``` 
the only thing u need to do so, is by passing `$config` variable to `WebSocketClient` class

**Comment by arthurkushman** [comment]
You can see the logic op changing flow in `WscMain::connect()` method that just apply those settings represented above, no need to make any sdk-calls, ex.:

https://github.com/arthurkushman/php-wss/blob/6d3dc0f5d0b9393972be85ed555e789b79c74b90/src/Components/WscMain.php#L95

And if proxy set correctly will call this function respectfully:
https://github.com/arthurkushman/php-wss/blob/6d3dc0f5d0b9393972be85ed555e789b79c74b90/src/Components/WscMain.php#L158
